src/persist-client/src/internal/compact.rs

```diff
@@ -710,7 +710,7 @@ where
             metrics.compaction.not_all_prefetched.inc();
         }
 
-        while let Some(updates) = consolidator.next().await {
+        while let Some(updates) = consolidator.next().await? {
```

Comment:
this punctuation mark is doing a lot of work! amazing how that changes the behavior here